### PR TITLE
bugfix: iife is 100% backwards compatible

### PR DESCRIPTION
**Pull request by givanse**
- [x] ember-cli-pretender can be bumped with no code changes https://github.com/rwjblue/ember-cli-pretender/pull/70
 - [x] ember-cli-mirage can ditch the shim and use the ES6 output https://github.com/samselikoff/ember-cli-mirage/pull/1503

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -1,37 +1,66 @@
...
-        'whatwg-fetch': 'FakeFetch',
-        'fake-xml-http-request': 'FakeXMLHttpRequest',
-        'route-recognizer': 'RouteRecognizer',
-      },
-      banner: 'var FakeFetch = self.WHATWGFetch;\n' +
-              'var FakeXMLHttpRequest = self.FakeXMLHttpRequest;\n' +
-              'var RouteRecognizer = self.RouteRecognizer;\n',
+      globals: globals,
+      banner: iife.banner,
+      footer: iife.footer,
```
Comment (givanse) [inline rollup.config.js]: I'm using an object to extract the iife's header and footer to reduce the noise in the rollup config.

File: rollup.config.js [hunk 2]
```diff
@@ -1,37 +1,66 @@
...
+  var FakeFetch = appearsBrowserified ? getModuleDefault(require('whatwg-fetch')) : self.WHATWGFetch;
+`,
+  footer:
+`
+if (typeof module === 'object') {
+  module.exports = Pretender;	
+} else if (typeof define !== 'undefined') {	
+  define('pretender', [], function() {	
+    return Pretender;	
+  });
```
Comment (givanse) [inline rollup.config.js]: This is the AMD module `define` that ember-cli-pretender depends on. Its not spec compliant, but that is intentional. (https://github.com/ember-cli/loader.js/commit/ce27cff63c10380559b49ef9979ca80619bc5fbb)

File: package.json [hunk 3]
```diff
@@ -54,6 +54,10 @@
     "route-recognizer": "^0.3.3",
     "whatwg-fetch": "^3.0.0"
   },
+  "files": [
+    "dist",
+    "index.d.ts"
+  ],
```
Comment (givanse) [inline package.json]: `npm publish --dry-run`

```
npm notice 📦  pretender@2.1.0
npm notice === Tarball Contents === 
npm notice 2.1kB  package.json        
npm notice 2.6kB  CHANGELOG.md        
npm notice 1.0kB  index.d.ts          
npm notice 1.1kB  LICENSE             
npm notice 13.0kB README.md           
npm notice 16.4kB dist/pretender.es.js
npm notice 18.5kB dist/pretender.js
```

File: rollup.config.js (hunk 2 above)
Comment (mike-north) [reply rollup.config.js]: I've refactored this out into an independent JS file for ease of maintenance

File: rollup.config.js (hunk 1 above)
Comment (mike-north) [reply rollup.config.js]: through refactoring the banner/footer into a JS file we just read and split at a special comment, this ends up being cleaned up even more!

File: .npmignore [hunk 4]
```diff
@@ -1,6 +0,0 @@
-src/index.ts
-test
-rollup.config.js
-karma.conf.js
-*.md
-yarn.lock
```
[the remark this answers is not in the document]
Comment (givanse) [reply .npmignore]: Yeah, intentional, `.npmignore` (black list) is harder to maintain than to use the `files` property in `package.json` (white list).

File: rollup.config.js [hunk 5]
```diff
@@ -1,42 +1,33 @@
...
+const fs = require('fs');
+const globals = {
+  'whatwg-fetch': 'FakeFetch',
+  'fake-xml-http-request': 'FakeXMLHttpRequest',
+  'route-recognizer': 'RouteRecognizer'
+};
 
-const selfId = path.resolve(__dirname, 'src/iife-self-placeholder.js');
+const rollupTemplate = fs.readFileSync('./iife-wrapper.js').toString();
+const [ banner, footer ] = rollupTemplate.split('/*==ROLLUP_CONTENT==*/');
```
Comment (givanse) [inline rollup.config.js]: neat 👌 

I'll verify these changes with the other repos and report back.

**Comment by givanse** [comment]
The other PRs (mirage, e-cli-pretender) still work 👍 Good to merge.

**Pull request merged**